assets/src/components/form/select/build-options.js

```diff
@@ -22,21 +23,36 @@ const buildOptions = (
 	optionsEntityMap,
 	mapSelection = 'default'
 ) => {
-	if ( isEmpty( entities ) || isEmpty( optionsEntityMap ) ) {
+	if ( isEmpty( entities ) ) {
+		warning(
+			false,
+			'A valid array of entities must be supplied ' +
+			'in order to build options for a ModelSelect component',
+			entities
+		);
+		return [];
```

Comment:
I don't think you should show a warning when entities are empty.  It's entirely feasible that a REST query could result in an empty set returned.  So you can check for empty entities and return and empty array right away but no need for warning imo.